### PR TITLE
FIX: The cjs template does not use MVC architecture pattern yet

### LINKED ISSUE #8: The cjs template does not use MVC architecture pattern yet
#### Issue description
The cjs template does not use the MVC architecture pattern yet i.e. Models, routes, services and controllers code are not yet separated into respective files. It makes it hard to read, and also hard to see similarities between the other 2 templates.

#### Steps to reproduce the issue
1. Go to https://github.com/code-collabo/node-mongo-api-boilerplate-templates
2. Compare cjs template's api folder content with that of ts template.
3. You will see that the ts template uses the MVC architecture successfully, but cjs is not using it. 

#### Expected behavior
The cjs template should use the MVC architecture, in order to look consistent with the ts template. But should retain the use of .then() syntax and make use of regular functions. 

#### Screenshots
None

#### Additional context
None

#### Testing checklist
Note: Add the testing checklist below (without this note) to your pull request when submitting your fix. This will guide (you and) the contributors who will review your fix to know what is important to check or test for. 
- [ ] all GET, POST, PATCH, PUT and DELETE requests have been restructed to use MVC: Models, routes, services and controllers are separated and in different files.
- [ ] Retains the use of .then() syntax and makes use of regular functions. 
- [ ] The function names and variable names are same as the ones found in the routes, controllers and services files of the ts template.
- [ ] No new code or functionality was added, I only restructured existing code to use MVC pattern. 
- [ ] I certify that I ran my checklist

### PR DESCRIPTION
**Pull request by samuko-things**
**This pull request makes the following changes:**
* Fixes code-collabo/node-mongo#8 

**General checklist**
- [x] all GET, POST, PATCH, PUT and DELETE requests have been restructed to use MVC: Models, routes, services and controllers are separated and in different files.
- [x] Retains the use of .then() syntax and makes use of regular functions.
- [x] The function names and variable names are same as the ones found in the routes, controllers and services files of the ts template.
- [x] No new code or functionality was added, I only restructured existing code to use MVC pattern.
- [x] I certify that I ran my checklist

Ping @code-collabo/node-mongo-api-boilerplate-templates

### CODE REVIEW

**Comment by samuko-things** [comment]
I have just restructured the cjs api template. do check it out

File: cjs/src/api/services/demo.service.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+//       });
+//     })
+//     .catch((err) => {
+//       res.status(500).json({
+//         message: `Error updating ${item}`,
+//         error: `${err}`,
+//       });
+//       console.log(chalk.redBright(`\nError updating ${item}: ${err}\n`));
+//     });
+// });
```
Comment (Ifycode) [inline cjs/src/api/services/demo.service.js]: @samuko-things Nice work. I haven't tested that it works yet. But you should also restructure patch and put to mvc too.

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: cjs/src/api/services/demo.service.js (hunk 1 above)
Comment (Ifycode) [reply cjs/src/api/services/demo.service.js]: Also note: 
- The space before the #8 does not make it link your PR to the issue properly. Please link the issue

**Comment by samuko-things** [comment]
i have just restructured the patch and put, do check it out.

File: cjs/src/api/controllers/demo.controller.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    .catch((err) => {
+      res.status(500).json({
+        message: `Error deleting ${item}`,
+        error: `${err}`,
+      });
+      console.log(chalk.redBright(`\nError deleting ${item}: ${err}\n`));
+    });
+};
+
+const partialUpdateDemoItemController = (req, res, next) => {
```
Comment (Ifycode) [inline cjs/src/api/controllers/demo.controller.js]: - Same thing for the cjs template. Change the function names accordingly and rearrange as I have said in Review comments 1, 2 & 3 in #4

**Review by Ifycode — CHANGES_REQUESTED** [review]

**Comment by samuko-things** [comment]
i have renamed the patch and put function as requested. do check it out

**Comment by samuko-things** [comment]
i have updated the cjs template api and also updated the app.js file to track the API routes

**Comment by samuko-things** [comment]
i have also tested it with atlas, and its working well. it also has a few warnings and error (which does not affect its functionality).

**Comment by Ifycode** [comment]
Thank you @samuko-things
I will check it out.
But I can't do that until you have corrected the prettier mess in the ts template PR. Once you have done that, I will review all 3 PRs submitted so far.

**Review by Ifycode — APPROVED** [review]

**Pull request merged**